### ISSUE TITLE
notice & alerts are not consistent across OS/Browser/screen reader

### ISSUE THREAD

**Issue by johrstrom** [OSC/ondemand#2077]
This notice that say a job was successfully started, does not alert screen readers. Indeed it would be very hard for a screen-reader user to find these notices.

I'd guess this applies to this success notice
https://github.com/OSC/ondemand/blob/26e1eee832bbe691163c296893efe1e627087d14/apps/dashboard/app/views/layouts/application.html.erb#L95-L96

as well as 
https://github.com/OSC/ondemand/blob/26e1eee832bbe691163c296893efe1e627087d14/apps/dashboard/app/views/layouts/application.html.erb#L85-L86

This one here above those both - I don't know and probably needs to be made into a modal anyhow.
https://github.com/OSC/ondemand/blob/26e1eee832bbe691163c296893efe1e627087d14/apps/dashboard/app/views/layouts/application.html.erb#L75

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1202821104799596/1202823733607536) by [Unito](https://www.unito.io)

**Comment by moffatsadeghi** [comment]
If it is done, we keep it and close it.  If this is not complete, we kick this to 4.2 / next release.

**Comment by Bubballoo3** [comment]
When I start a session the `div.alert.alert-success.alert-dismissible` that contains "app was successfully created" gets read. The message for job completion gets read as well, although that goes to the #aria_live_region element. 

If there are submission errors that keep you on the form, these are not read. When you delete a session and the "successfully deleted" replaces "successfully created", this update is not read. 

As an aside, the "js-alert-danger-template" element is hidden from view, and anything that uses it should probably use the #aria_live_region instead.

Hopefully this gives us a good starting point when we circle back to this in 4.2

**Comment by Bubballoo3** [comment]
After looking into this again, there does not seem to be any meaningful structural difference between
1. reloading after a failed submission gives a 'This app does not supply a sub app form file under the directory' error that is read by screenreaders, triggered by `flash.now[:alert]`.
2. When the app submission fails, we do not use 'flash.now', but just insert the error message into the HTML. While the errors go into different divs, both are marked with `role='alert'`, which should include `aria-live` functionality. 

These are the two most similar examples to demonstrate this behavior; possible fixes could be transitioning to `flash.now` for all of these, though this would sacrifice some of the structure we are currently able to show in the second case.

**Comment by johrstrom** [comment]
Apparently this is a Firefox issue. Chrome does in fact read the alerts as it should. I tested with Windows/NVDA/FF and Windows/NVDA/Chrome combinations.

**Comment by Bubballoo3** [comment]
I just tested on both safari with VoiceOver, and Edge and Chrome with NVDA. All of them start over at the top of the page after a failed submission. The alert looks like 

<img width="1844" height="392" alt="Image" src="https://github.com/user-attachments/assets/a551acfa-c274-46cb-93d3-66229f313a05" />
 
and the NVDA output (with chrome) is the following (the launch button at the top is immediately clicked, submitting the form)
```
clickable  Launch  button  
Loading directory contents 
Mathematica - OSC OnDemand
clickable  banner landmark  clickable    same page  link    Skip Navigation
navigation landmark  clickable    menu bar  clickable    menu item    OSC OnDemand  clickable  menu item  subMenu    Files
menu item  subMenu    clickable  Jobs
menu item  subMenu    clickable  Clusters
Directory loaded with 30 items 
menu item  subMenu    clickable  Interactive Apps
menu item    clickable  My Interactive Sessions
clickable
menu item  subMenu    clickable   Develop
menu item  subMenu    clickable   Help
If this job failed to submit because of an invalid job name please ask your administrator to configure OnDemand to set the environment variable OOD_JOB_NAME_ILLEGAL_CHARS.
sbatch: error: Invalid numeric value "0" for --ntasks-per-node.
```

**Comment by johrstrom** [comment]
Yea they have the `role="alert"` though. I assume you tested Chrome on MacOS?

**Comment by johrstrom** [comment]
I played around with focusing the element. It seems like it shouldn't be necessary, but I'm at a loss as to what else can be done. I mean the div has the right role. Maybe playing with the content inside to be text only can help (other than headers and lists), but that's a maybe.

**Comment by johrstrom** [comment]
I was testing on Windows 11 `10.0.22631 Build 22631` and Chrome `Version 147.0.7727.56 (Official Build) (64-bit)`

**Comment by johrstrom** [comment]
I'm trying to compile a complete table of behavior as I found 4.1 works with some combinations of SR and browser. I've only been able to make it through 4.1 testing without any patches yet. At first, it just seems like an issue for Firefox and Windows Narrator. I generally use Firefox, so that's how I found the issue to begin with.


| OS      | Browser | ScreenReader   |  OOD | BC Failure Alert  |  BC Success Notice  |
| :-------: | :--------: | :--------: | :--------: | :--------: | :-------: |
| Windows 11     | Chrome       | NVDA  | 4.1 | works | works |
| Windows 11     | Edge       | NVDA  | 4.1 | works | works |
| Windows 11     | FireFox       | NVDA  | 4.1 | doesn't work | doens't work |
| Windows 11     | Chrome       | Windows Narrator  | 4.1 | half works with no 'alert' | half works with no 'alert' |
| Windows 11     | Edge       | Windows Narrator  | 4.1 | works with no 'alert' | works with no 'alert' |
| Windows 11     | FireFox       | Windows Narrator  | 4.1 | doens't work | doesn't work |
| Windows 11     | Chrome       | NVDA  | cc6b | works | works |
| Windows 11     | Edge       | NVDA  | cc6b | works | works |
| Windows 11     | FireFox       | NVDA  | cc6b | works |works |
| Windows 11     | Chrome       | Windows Narrator  | cc6b | half works with no 'alert' | half works with no 'alert' |
| Windows 11     | Edge       | Windows Narrator  | cc6b | works with no 'alert' | works with no 'alert' |
| Windows 11     | FireFox       | Windows Narrator  | cc6b | doesn't work | doesn't work |
| Windows 11     | Chrome       | NVDA  | af80 | works | works |
| Windows 11     | Edge       | NVDA  | af80 | works | works |
| Windows 11     | Firefox       | NVDA  | af80 | works | works |
| Windows 11     | Chrome       | Windows Narrator  | af80 | half works with no 'alert' | half works with no 'alert' |
| Windows 11     | Edge       | Windows Narrator  | af80 | works with no 'alert' | half works with no 'alert' |
| Windows 11     | Firefox       | Windows Narrator  | af80 | doesn't work  | doesn't work |

* "half works" means it's reads some of the navbar, then starts to read the alert/notice.
* "no 'alert'" just means it reads the text, but does not preface the text with 'alert'

**Comment by johrstrom** [comment]
OK - completed the table up to https://github.com/OSC/ondemand/pull/5322/changes/cc6b917b9fcb3bc11b9c8750a57b33ac3719dd22 . Seems like MacOS is consistently in the `half works with no 'alert'` category?

**Comment by Bubballoo3** [comment]
Yeah I'd say so. It half works most of the time, and will inconsistently not work at all (ie, finish reading through the navbar before getting to the alert in the standard order)

**Comment by Bubballoo3** [comment]
No regression on mac between 4.1 and cc6b. Wildly inconsistent on both, but seems to have similar chances of half working on both.

**Comment by johrstrom** [comment]
I've made a new commit to the PR to hide the content on page load, then update the HTML element in javascript to generate an alert and show the alert visually.

Though it seems all we really get is FF/NVDA/Windows 11 support. Windows narrator has the same behavior as 4.1.
